### PR TITLE
Core/Totem: Immune to all positive spells.

### PR DESCRIPTION
**Pull request by HoustonCore**
**Changes proposed:**

- Immune to all positive spells, except of stoneclaw totem absorb and sentry totem bind sight
- Totems positive spells have unit_caster target

**Target branch(es):** 3.3.5/master

- [√ ] 3.3.5

**Tests performed:**

Does it build, tested in-game, etc.

**Tests:**
[Look video](https://youtu.be/qw6ySrBLgho)

### CODE REVIEW

File: src/server/game/Entities/Totem/Totem.cpp [hunk 1]
```diff
@@ -157,9 +157,13 @@ void Totem::UnSummon(uint32 msTime)
 
 bool Totem::IsImmunedToSpellEffect(SpellInfo const* spellInfo, uint32 index, WorldObject const* caster) const
 {
-    /// @todo possibly all negative auras immune?
-    if (GetEntry() == 5925)
-        return false;
+    // immune to all positive spells, except of stoneclaw totem absorb and sentry totem bind sight
+	// totems positive spells have unit_caster target
```
Comment (Faq) [inline src/server/game/Entities/Totem/Totem.cpp:161]:
```suggestion
    // totems positive spells have unit_caster target
```

File: src/server/game/Entities/Totem/Totem.cpp [hunk 2]
```diff
@@ -157,9 +157,13 @@ void Totem::UnSummon(uint32 msTime)
...
+    if (spellInfo->Effects[index].Effect != SPELL_EFFECT_DUMMY && 
+        spellInfo->Effects[index].Effect != SPELL_EFFECT_SCRIPT_EFFECT && 
+        spellInfo->IsPositive() && spellInfo->Effects[index].TargetA.GetTarget() != TARGET_UNIT_CASTER &&
+        spellInfo->Effects[index].TargetA.GetCheckType() != TARGET_CHECK_ENTRY && spellInfo->Id != 55277 && spellInfo->Id != 6277)
```
Comment (Faq) [inline src/server/game/Entities/Totem/Totem.cpp:165]: avoid using magic numbers, like 55277, 6277

File: src/server/game/Entities/Totem/Totem.cpp (hunk 1 above)
Comment (MokiGitHub) [reply src/server/game/Entities/Totem/Totem.cpp]: Done

File: src/server/game/Entities/Totem/Totem.cpp (hunk 2 above)
Comment (MokiGitHub) [reply src/server/game/Entities/Totem/Totem.cpp:165]: > avoid using magic numbers, like 55277, 6277

Do I declare them in Totem.h? I'm just learning)

**Comment by Jildor** [comment]
any proof about that totems should be immune?

**Comment by MokiGitHub** [comment]
> any proof about that totems should be immune?

Edit: Try to google "buffing (or healing) shaman totem wow" there are zero instances where this occured on retail, you will not find anything relevant. If you ever engaged a shaman who used Fire Elemental Totem you would always kill the totem rather than the elemental because the totem couldn't be healed while the elemental could. If this was a thing you could just put Earth Shield on the totem to protect it from being killed.
Source: my 350 days /played as a shaman in TBC/Wotlk/Cata

Yes, and I know some servers have issues with Chain Heal jumping to the totems. Couldn't reproduce it here but it's something to keep in mind. Chain Heal and Prayer of Mending should ignore them as targets.

P.S. This commentnot my.

File: src/server/game/Entities/Totem/Totem.cpp (hunk 2 above)
Comment (MokiGitHub) [reply src/server/game/Entities/Totem/Totem.cpp]: Done, check pls.

**Comment by MokiGitHub** [comment]
Can someone test it?

**Comment by Shauren** [comment]
Yeah totems are immune to being buffed and healing as well, this is still the case even in bfa

**Comment by jackpoz** [comment]
are there any quest spells that might be affected by this change ?

**Comment by MokiGitHub** [comment]
> are there any quest spells that might be affected by this change ?

I think no.

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**